wideq/core.py

```diff
@@ -1,183 +1,70 @@
 """A low-level, general abstraction for the LG SmartThinQ API.
 """
-import base64
-import uuid
+import ssl
+import requests
+from urllib3.poolmanager import PoolManager
+from requests.adapters import HTTPAdapter
 from urllib.parse import urljoin, urlencode, urlparse, parse_qs
+import base64
 import hashlib
 import hmac
-import datetime
-import requests
 import logging
-from typing import Any, Dict, List, Tuple
-from requests.adapters import HTTPAdapter
-from requests.packages.urllib3.util.retry import Retry
-
-GATEWAY_URL = 'https://kic.lgthinq.com:46030/api/common/gatewayUriList'
-APP_KEY = 'wideq'
-SECURITY_KEY = 'nuts_securitykey'
-DATA_ROOT = 'lgedmRoot'
-SVC_CODE = 'SVC202'
-CLIENT_ID = 'LGAO221A02'
-OAUTH_SECRET_KEY = 'c053c2a6ddeb7ad97cb0eed0dcb31cf8'
-OAUTH_CLIENT_KEY = 'LGAO221A02'
-DATE_FORMAT = '%a, %d %b %Y %H:%M:%S +0000'
-DEFAULT_COUNTRY = 'US'
-DEFAULT_LANGUAGE = 'en-US'
-
-RETRY_COUNT = 5  # Anecdotally this seems sufficient.
-RETRY_FACTOR = 0.5
-RETRY_STATUSES = (502, 503, 504)
-
-
-def get_wideq_logger() -> logging.Logger:
-    level = logging.INFO
-    fmt = "%(asctime)s %(levelname)s [%(name)s] %(message)s"
-    datefmt = "%Y-%m-%d %H:%M:%S"
-    logger = logging.getLogger("wideq")
-    logger.setLevel(level)
-
-    try:
-        import colorlog  # type: ignore
-        colorfmt = f"%(log_color)s{fmt}%(reset)s"
-        handler = colorlog.StreamHandler()
-        handler.setFormatter(
-            colorlog.ColoredFormatter(
-                colorfmt,
-                datefmt=datefmt,
-                reset=True,
-                log_colors={
-                    "DEBUG": "cyan",
-                    "INFO": "green",
-                    "WARNING": "yellow",
-                    "ERROR": "red",
-                    "CRITICAL": "red",
-                },
-            )
-        )
-    except ImportError:
-        handler = logging.StreamHandler()
-        handler.setFormatter(logging.Formatter(fmt=fmt, datefmt=datefmt))
-
-    logger.addHandler(handler)
-    return logger
+from datetime import datetime
+from typing import Any, Dict, Generator, Optional
 
+from . import as_list, gen_uuid
+from . import core_exceptions as exc
+from .device import DeviceInfo, DEFAULT_TIMEOUT, DEFAULT_REFRESH_TIMEOUT
 
-LOGGER = get_wideq_logger()
 
+class Tlsv1HttpAdapter(HTTPAdapter):
```

Comment:
Is this necessary? I know some people get TLS errors when they use the wrong country server, but reverting to TLSv1 for all connections is a security concern. (Maybe we can leave this out for now and focus on other stuff first?)

Reply:
This is more related to some country server that need this.
Probably some check should be done during correction to choose the right TLS connection, otherwise for some user do not work

Reply:
Great; that seems like a good idea. For this first iteration, let's leave stuff like this out and just focus on the minimal, core changes to support version 2 of the LG API.